src/main/java/com/igot/cb/masterdata/service/MasterDataServiceImpl.java

```diff
@@ -512,4 +555,358 @@ protected boolean updateOrAddDegree(List<String> degrees, String degreeName) {
+                return apiResponse;
+            }
+            apiResponse.getResult().put(Constants.RESULT, esResponse.getData());
+            apiResponse.getResult().put(Constants.COUNT, esResponse.getCount());
+        } catch (IllegalArgumentException e) {
+            logger.error("Invalid pagination or sorting parameters: {}", e.getMessage(), e);
+            ProjectUtil.errorResponse(apiResponse, "Invalid pagination or sorting parameters", HttpStatus.BAD_REQUEST);
+
+        } catch (Exception e) {
+            logger.error("Unexpected error during institute search: {}", e.getMessage(), e);
+            ProjectUtil.errorResponse(apiResponse, "Unexpected error during institute search", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        return apiResponse;
+    }
+
+    public ApiResponse addDegree(Degree degree) {
+        ApiResponse apiResponse = ProjectUtil.createDefaultResponse(Constants.API_ADD_DEGREE);
+
+        try {
+            // ------------------ Basic Validation ------------------
+            if (StringUtils.isEmpty(degree.getName())) {
+                ProjectUtil.errorResponse(apiResponse, "Degree name cannot be empty", HttpStatus.BAD_REQUEST);
+                return apiResponse;
+            }
+            if (StringUtils.isNotEmpty(degree.getDescription()) && degree.getDescription().length() > 255) {
+                ProjectUtil.errorResponse(apiResponse, "Degree description cannot exceed 255 characters", HttpStatus.BAD_REQUEST);
+                return apiResponse;
+            }
+            // ------------------ Check Existing ------------------
+            Optional<Degree> existingDegreeOpt = degreeRepository.findByNameIgnoreCase(degree.getName());
+            if (existingDegreeOpt.isPresent()) {
+                Degree existingDegree = existingDegreeOpt.get();
+                if (existingDegree.getStatus() == 0) {
+                    // Reactivate inactive record in Postgres
+                    existingDegree.setStatus(1);
+                    if (StringUtils.isNotEmpty(degree.getDescription())) {
+                        existingDegree.setDescription(degree.getDescription());
+                    }
+                    Degree updated = degreeRepository.save(existingDegree);
+                    // Sync ES
+                    try {
+                        EsResponse esResponse = esUtilService.saveObjectInIgotES(updated, serverProperties.getEsDegreeIndexName(), serverProperties.getEsMasterDataIndexDocType(), updated.getId().toString());
+                        if (!esResponse.isSuccess()) {
+                            logger.warn("Failed to index reactivated degree in ES: {}", esResponse.getMessage());
+                        }
+                    } catch (Exception e) {
+                        logger.warn("Exception while indexing reactivated degree in ES: {}", e.getMessage(), e);
+                    }
```

Comment:
move these to different class... 
this try block is inside of another try block -- which will raise a sonarQube issues